python/lib/communication/dmod/communication/evaluation_request.py

```diff
@@ -0,0 +1,163 @@
+import typing
+import abc
+import json
+
+from numbers import Number
+from typing import Dict
+from typing import Union
+
+from .maas_request import MessageEventType, Response
+from .message import Message
+
+SERIALIZABLE_DICT = typing.Dict[str, typing.Union[str, Number, dict, typing.List]]
+
+
+class EvaluationRequest(Message, abc.ABC):
+    """
+    A request to be forwarded to the evaluation service
+    """
+
+    event_type: MessageEventType = MessageEventType.EVALUATION_REQUEST
+    """ :class:`MessageEventType`: the event type for this message implementation """
+
+    @classmethod
+    @abc.abstractmethod
+    def get_action(cls) -> str:
+        ...
+
+    @property
+    def action(self) -> str:
+        return self.get_action()
+
+
+class EvaluationConnectionRequest(EvaluationRequest):
+    """
+    A request used to communicate through a chained websocket connection
+    """
+    _action_parameters: typing.Dict[str, typing.Any]
+
+    def __init__(self, **kwargs):
+        self._action_parameters = kwargs or dict()
+
+    @classmethod
+    def get_action(cls) -> str:
```

Comment:
I *think* so? I don't see *where* it might lead things off the rails. Should I switch it up prior to acceptance or should we just keep an eye out for it and any possible similar situations?

Reply:
I think it is fine. Given how classmethods are used elsewhere in the codebase, I think I am just getting use to how we are using them.